frontend/src/ui/Access.tsx

```diff
@@ -336,46 +371,97 @@ const AclSelect: React.FC<AclSelectProps> = ({ acl, kind }) => {
                 },
             }}>
                 <colgroup>
-                    <col css={{ }} />
-                    <col css={{ width: i18n.resolvedLanguage === "en" ? 190 : 224 }} />
+                    <col />
+                    <col css={{
+                        // This is a little hacky and kind of messy.
+                        // We want the width of the action columns to be fixed
+                        // and determined by the longest action label (e.g. `read and write`
+                        // should also determine the width of the `read` label). These lengths are
+                        // of course also dependent on the current language (german for example
+                        // has a reputation for having ridiculously lengthy words).
+                        // Now that we use this selector also for realm acl that use different
+                        // labels altogether, this needs yet another check.
+                        // This will of course become even more complicated once more languages are
+                        // added.
+                        width: permissionLevels.highest === "admin"
+                            ? i18n.resolvedLanguage === "en" ? 155 : 160
+                            : i18n.resolvedLanguage === "en" ? 190 : 224,
+                    }} />
                     <col css={{ width: 42 }} />
                 </colgroup>
                 <thead>
                     <tr css={{
                         borderBottom: `2px solid ${COLORS.neutral05}`,
                         textAlign: "left",
                     }}>
-                        <th>{translations.columnHeader}</th>
+                        <th>{t(`manage.access.table.${kind}`)}</th>
                         <th>{t("manage.access.table.actions.title")}</th>
                         <th></th>
                     </tr>
                 </thead>
                 <tbody>
                     {/* Placeholder if there are no entries */}
-                    {entries.length === 0 && !userIsAdmin && <tr>
-                        <td colSpan={3} css={{ textAlign: "center", fontStyle: "italic" }}>
-                            {t("acl.no-entries")}
-                        </td>
-                    </tr>}
+                    {
+                        entries.length === 0
+                        && !(showAdminEntry || showUserEntry)
+                        && <tr>
+                            <td colSpan={3} css={{ textAlign: "center", fontStyle: "italic" }}>
+                                {t("acl.no-entries")}
+                            </td>
+                        </tr>
+                    }
```

Comment:
I would think the previous layout of `{condition && <tr>\n...` would still work?